src/main/java/org/dependencytrack/resources/v1/TeamResource.java

```diff
@@ -220,6 +224,38 @@ public Response deleteTeam(Team jsonTeam) {
         }
     }
 
+    @GET
+    @Path("/visible")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Returns a list of Teams that are visible", description = "<p></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "The Visible Teams", content = @Content(array = @ArraySchema(schema = @Schema(implementation = VisibleTeams.class)))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized")
+    })
+    @PermissionRequired({Permissions.Constants.ACCESS_MANAGEMENT, Permissions.Constants.ACCESS_MANAGEMENT_READ})
```

Comment:
so we won't be needing the below check as well in the endpoint?
`boolean isAllTeams = qm.hasAccessManagementPermission(user);`

Reply:
That is still required. Basically, the endpoint is supposed to tell a user which teams are visible to them. So if they're member of team `A`, but not of team `B`, the endpoint will only return team `A`.

Users with `ACCESS_MANAGEMENT` permission can see *all* teams, no matter if they're a member or not. That's because the `ACCESS_MANAGEMENT` permission allows them to manage teams and memberships anyway.